pkg/secretless/config/v2/config_http.go

```diff
@@ -36,49 +43,45 @@ func newHTTPConfig(cfgBytes []byte) (*httpConfig, error) {
 }
 
 func (cfg *httpConfig) UnmarshalYAML(bytes []byte) error {
-	err := yaml.Unmarshal(bytes, cfg)
-
-	// A string type is converted into [] by default, so we must verify length.
-	// If this passes, all is good and we can return.
-	if ok := err == nil && len(cfg.AuthenticateURLsMatching) > 0; ok {
-		return nil
-	}
-
-	// Unmarshall into a temp struct that expects a string
+	// Unmarshall into a temp struct
+	//
+	// This temp struct makes it possible to parse 'authenticateURLsMatching' as
+	// string or []string
 	tempCfg := &struct {
-		AuthenticationStrategy   string `yaml:"authenticationStrategy"`
-		AuthenticateURLsMatching string `yaml:"authenticateURLsMatching"`
+		AuthenticateURLsMatching interface{} `yaml:"authenticateURLsMatching"`
 	}{}
-	err = yaml.Unmarshal(bytes, tempCfg)
-
-	// It must succeed with a non-empty string
-	if ok := err == nil && len(tempCfg.AuthenticateURLsMatching) > 0; !ok {
-		return errors.New("http ProtocolConfig could not be parsed")
+	err := yaml.Unmarshal(bytes, tempCfg)
+	if err != nil {
+		return errors.New("http ConnectorConfig could not be parsed")
 	}
 
-	// It's a string, let's convert it to a []string
-	cfg.AuthenticationStrategy = tempCfg.AuthenticationStrategy
-	cfg.AuthenticateURLsMatching = []string{ tempCfg.AuthenticateURLsMatching }
+	// Populate actual http config from tempCfg
+	switch v := tempCfg.AuthenticateURLsMatching.(type) {
+	case string:
+		cfg.AuthenticateURLsMatching = []string{ v }
+	case []interface{}:
+		urlMatchStrings := make([]string, len(v))
+		for urlMatchIndex, urlMatch := range v {
```

Comment:
I'd change `urlMatchIndex` -> `i`.  No need to go against convention unless you have a really long loop or something.